Support pay as you go scalardl in aws marketplace (#198)
scalar-labs/scalar-kubernetes: docs/AwsMarketplaceGuide.md

# How to install Scalar products through AWS Marketplace

> Scalar products (ScalarDB, ScalarDL, and their tools) are provided in AWS Marketplace as Container Image. This guide explains how to install Scalar products through AWS Marketplace.

Reviewer: ```suggestion
Scalar products (ScalarDB, ScalarDL, and their tools) are available in the AWS Marketplace as container images. This guide explains how to install Scalar products through the AWS Marketplace.
```

> Note that some Scalar products are licensed under commercial licenses, and the AWS Marketplace provides them as Pay-As-You-Go and BYOL (Bring Your Own License). If you select Pay-As-You-Go, AWS charges you our product license fee. If you select BYOL, please make sure you have appropriate licenses.

Author: Ah, yes. I think it is better to say `based on your usage`. I will update this sentence based on your suggestion! Thank you!

1. Create an OIDC provider.

   You must create an identity and access management (IAM) OpenID Connect (OIDC) provider to run the AWS Marketplace Metering Service from ScalarDL pods.

   ```console
   eksctl utils associate-iam-oidc-provider --region <REGION> --cluster <EKS_CLUSTER_NAME> --approve
   ```

   For details, see [Creating an IAM OIDC provider for your cluster](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html).

1. Create a service account.

   To allow your pods to run the AWS Marketplace Metering Service, you can use [IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).

   ```console
   eksctl create iamserviceaccount \
       --name <SERVICE_ACCOUNT_NAME> \
       --namespace <NAMESPACE> \
       --region <REGION> \
       --cluster <EKS_CLUSTER_NAME> \
       --attach-policy-arn arn:aws:iam::aws:policy/AWSMarketplaceMeteringFullAccess \
       --approve \
       --override-existing-serviceaccounts
   ```

1. Update the custom values file of the Helm Chart of a Scalar product you want to install.

>    You need to specify the private container registry (ECR) of AWS Marketplace and the version (tag) as the value of `[].image.repository` and `[].image.version (tag)` in the custom values file. You also need to specify the service account name that you created in the previous step as the value of `[].serviceAccount.serviceAccountName` and set `true` to `[].serviceAccount.automountServiceAccountToken`.

Reviewer: ```suggestion
   You need to specify the private container registry (ECR) of the AWS Marketplace and the version (tag) as the value for `[].image.repository` and `[].image.version (tag)` in the custom values file. You also need to specify the service account name that you created in the previous step as the value for `[].serviceAccount.serviceAccountName` and set `[].serviceAccount.automountServiceAccountToken` to `true`.
```

            version: "3.8.1"
          serviceAccount:
            serviceAccountName: "<SERVICE_ACCOUNT_NAME>"
            automountServiceAccountToken: true
        ```
      * ScalarDL Auditor (scalardl-auditor-custom-values.yaml)
        ```yaml
        auditor:
          image:
            repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-auditor-aws-payg"
            version: "3.8.1"
          serviceAccount:
            serviceAccountName: "<SERVICE_ACCOUNT_NAME>"
            automountServiceAccountToken: true
        ```
      * ScalarDL Schema Loader for Ledger (schema-loader-ledger-custom-values.yaml)
        ```yaml
        schemaLoading:
          image:
            repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-schema-loader-ledger-aws-payg"
            version: "3.8.0"
        ```
      * ScalarDL Schema Loader for Auditor (schema-loader-auditor-custom-values.yaml)
        ```yaml
        schemaLoading:
          image:
            repository: "709825985650.dkr.ecr.us-east-1.amazonaws.com/scalar/scalardl-schema-loader-auditor-aws-payg"
            version: "3.8.0"
        ```

> 1. Deploy the Scalar products using the Helm Chart with the above custom values files.

Reviewer: ```suggestion
1. Deploy Scalar products by using Helm Charts in conjunction with the above custom values files.
```

   * ScalarDL Examples
      * ScalarDL Ledger
        ```console
        helm install scalardl-ledger scalar-labs/scalardl -f ./scalardl-ledger-custom-values.yaml
        ```
      * ScalarDL Auditor
        ```console
        helm install scalardl-auditor scalar-labs/scalardl-audit -f ./scalardl-auditor-custom-values.yaml
        ```
      * ScalarDL Schema Loader (Ledger)
        ```console
        helm install schema-loader scalar-labs/schema-loading -f ./schema-loader-ledger-custom-values.yaml
        ```
      * ScalarDL Schema Loader (Auditor)
        ```console
        helm install schema-loader scalar-labs/schema-loading -f ./schema-loader-auditor-custom-values.yaml
        ```

## **[BYOL]** Deploy containers on EKS (Amazon Elastic Kubernetes Service) from AWS Marketplace using Scalar Helm Charts

> If you subscribe to Scalar products in AWS Marketplace, you can pull the container images of Scalar products from the private container registry ([ECR](https://aws.amazon.com/ecr/)) of AWS Marketplace. This section explains how to deploy BYOL Scalar products on your [EKS](https://aws.amazon.com/eks/) cluster from the private container registry.

Reviewer: ```suggestion
By subscribing to Scalar products in the AWS Marketplace, you can pull the container images of Scalar products from the private container registry ([ECR](https://aws.amazon.com/ecr/)) of the AWS Marketplace. This section explains how to deploy Scalar products with the BYOL option in your [EKS](https://aws.amazon.com/eks/) cluster from the private container registry.
```
